oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java

```diff
@@ -1201,7 +1201,13 @@ private SelectorExecutionPlan getBestSelectorExecutionPlan(
         }
 
         if (potentiallySlowTraversalQuery || bestIndex == null) {
-            LOG.debug("no proper index was found for filter {}", filter);
+            // Log error for fulltext queries without index, since these cannot return results
+            if(!filter.getFulltextConditions().isEmpty()) { 
+                LOG.error("Fulltext query without index for filter {}; no results will be returned", filter);
```

Comment:
Nit: I think it should be "WARN" to be in-line with other messages (e.g. "The query read more than...") -- I would use errors for data corruption and so on only.

Reply:
I was unsure on this too; my thinking was that it should be an ERROR because the query will _never_ return results - so it's not that it's not performant, it's actually effectively failing.